src/impl_/ghost.rs

```diff
@@ -0,0 +1,19 @@
+/// If it does nothing, was it ever really there? 👻
+///
+/// This is code that is just type checked to e.g. create better compile errors,
+/// but that never affects anything at runtime,
+use crate::{IntoPy, PyErr, PyObject};
+
+#[allow(clippy::wrong_self_convention)]
```

Comment:
Is this lint actually necessary? I wonder if it's a hangover from experimentation.

Reply:
Ah I see! Do you think we should name the method differently then? E.g. `into_py_result_hint` or `assert_into_py_result`?